hatasmota/camera.py

```diff
@@ -0,0 +1,141 @@
+
+    @classmethod
+    def from_discovery_message(
+        cls, config: dict, platform: str
+    ) -> TasmotaCameraConfig:
+        """Instantiate from discovery message."""
+        return cls(
+            endpoint="camera",
+            idx=0,
+            friendly_name=None,
+            mac=config[CONF_MAC],
+            platform=platform,
+            poll_payload="",
+            poll_topic=get_topic_command_state(config),
+            availability_topic=get_topic_tele_will(config),
+            availability_offline=config_get_state_offline(config),
+            availability_online=config_get_state_online(config),
+            deep_sleep_enabled=config[CONF_DEEP_SLEEP],
+            command_topic=get_topic_command(config),
+            result_topic=get_topic_stat_result(config),
+            state_topic=get_topic_tele_state(config),
+            sensor_topic=get_topic_tele_sensor(config),
+            ip_address=config[CONF_IP]
+        )
+
+
+class TasmotaCamera(TasmotaAvailability, TasmotaEntity):
+    """Representation of a Tasmota camera."""
+
+    _cfg: TasmotaCameraConfig
+
+    def __init__(self, **kwds: Any):
+        """Initialize."""
+        self._sub_state: dict | None = None
+        super().__init__(**kwds)
+
+    async def subscribe_topics(self) -> None:
+        """Subscribe to topics."""
+
+        def state_message_received(msg: ReceiveMessage) -> None:
+            """Handle new MQTT state messages."""
+            if not self._on_state_callback:
+                return
+
+            if msg.topic == self._cfg.sensor_topic:
+                state = get_value_by_path(msg.payload, ["CAMERA"])
+                if state:
+                    self._on_state_callback(state)
+
+
+        availability_topics = self.get_availability_topics()
+        topics = {
+            "result_topic": {
+                "event_loop_safe": True,
+                "topic": self._cfg.result_topic,
+                "msg_callback": state_message_received,
+            },
+            "state_topic": {
+                "event_loop_safe": True,
+                "topic": self._cfg.state_topic,
+                "msg_callback": state_message_received,
+            },
+            "sensor_topic": {
+                "event_loop_safe": True,
+                "topic": self._cfg.sensor_topic,
+                "msg_callback": state_message_received,
+            },
+        }
+        topics = {**topics, **availability_topics}
+
+        self._sub_state = await self._mqtt_client.subscribe(
+            self._sub_state,
+            topics,
+        )
+
+    async def unsubscribe_topics(self) -> None:
+        """Unsubscribe to all MQTT topics."""
+        self._sub_state = await self._mqtt_client.unsubscribe(self._sub_state)
+
+    def get_still_image_stream(self, websession:aiohttp.ClientSession) -> web.StreamResponse | None:
+        """Get the io stream to read the static image."""
+        still_image_url = f"http://{self._cfg.ip_address}/snapshot.jpg"
```

Comment:
Is the URL always `/snapshot.jpg`, or can it be configured?

Reply:
From the [Tasmota code](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_81_esp32_webcam.ino#L1110) (or [new version](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_81_esp32_webcam_task.ino#2228)), we have a choice of `/snapshot.jpg`, `/wc.jpg`, `/wc.mjpeg`. But IMO, `snapshot.jpg` seems the best description of the functionality here.

Reply:
OK, but just to clarify: Is there a difference between `/snapshot.jpg`, `/wc.jpg` and `/wc.mjpeg`?

Reply:
There is no difference. All these URLs have the same callback function and return the same data.
I guess the Tasmota authors added them only for convenience.